lib/src/widgets/wolt_bottom_sheet_drag_handle.dart

```diff
@@ -11,32 +11,39 @@ class WoltBottomSheetDragHandle extends StatelessWidget {
   Widget build(BuildContext context) {
     final themeData = Theme.of(context).extension<WoltModalSheetThemeData>();
     final defaultThemeData = WoltModalSheetDefaultThemeData(context);
-    final handleSize =
-        themeData?.dragHandleSize ?? defaultThemeData.dragHandleSize;
-    final handleColor =
-        themeData?.dragHandleColor ?? defaultThemeData.dragHandleColor;
+    final handleSize = themeData?.dragHandleSize ?? defaultThemeData.dragHandleSize;
+    final handleColor = themeData?.dragHandleColor ?? defaultThemeData.dragHandleColor;
+
+    // Ensure that handle size does not exceed the minimum interactive dimension.
+    final adjustedHandleWidth = handleSize.width.clamp(0.0, _minInteractiveDimension);
+    final adjustedHandleHeight = handleSize.height.clamp(0.0, _minInteractiveDimension);
+
+    // Calculate padding to center the handle.
+    final horizontalPadding = (_minInteractiveDimension - adjustedHandleWidth) / 2;
+    const topPadding = 8.0;
+    final bottomPadding = _minInteractiveDimension - topPadding - adjustedHandleHeight;
+
     return Semantics(
       label: MaterialLocalizations.of(context).modalBarrierDismissLabel,
       container: true,
-      child: SizedBox.square(
-        dimension: _minInteractiveDimension,
-        // By setting behavior to HitTestBehavior.opaque, the GestureDetector will capture touch
-        // events even if its child (the drag handle) isn't the exact size of the gesture.
-        // This will prevent the scrollable content below from receiving the touch events,
-        // effectively allowing the handler to capture drag gestures.
-        child: GestureDetector(
-          behavior: HitTestBehavior.opaque,
-          child: Align(
-            alignment: Alignment.topCenter,
-            child: Container(
-              margin: const EdgeInsets.only(top: 8.0),
-              height: handleSize.height,
-              width: handleSize.width,
-              decoration: BoxDecoration(
-                borderRadius: BorderRadius.circular(handleSize.height / 2),
-                color: handleColor,
-              ),
+      child: GestureDetector(
+        behavior: HitTestBehavior.opaque,
+        child: SizedBox(
+          width: _minInteractiveDimension,
+          height: _minInteractiveDimension,
+          child: Container(
+            margin: EdgeInsets.fromLTRB(
+              horizontalPadding,
+              topPadding,
+              horizontalPadding,
+              bottomPadding.clamp(0.0, _minInteractiveDimension),
+            ),
+            decoration: BoxDecoration(
+              color: handleColor,
+              borderRadius: BorderRadius.circular(adjustedHandleHeight / 2),
             ),
+            width: adjustedHandleWidth,
+            height: adjustedHandleHeight,
```

Comment:
Can you please explain `SizedBox` with size on top of `Container` with a size?  
I feel like one of them shouldn't be required.

Reply:
Container is the decorated handle box itself. The SizedBox is the interactive area which is a transparent box.

Reply:
Thanks for the tip: converted to sized box here: 96d4bc8c51ead2f8694bb19d4415768fc937fec7